client/jetpack-connect/auth-logged-in-form.jsx

```diff
@@ -284,8 +284,11 @@ class LoggedInForm extends Component {
 	}
 
 	renderNotices() {
-		const { authorizeError, queryObject, isAuthorizing, authorizeSuccess } = this.props.jetpackConnectAuthorize;
-		if ( queryObject.already_authorized && ! this.props.isFetchingSites && ! this.props.isAlreadyOnSitesList ) {
+		const { authorizeError, queryObject, isAuthorizing, authorizeSuccess, userAlreadyConnected } = this.props.jetpackConnectAuthorize;
+		if (
+			( queryObject.already_authorized && ! this.props.isFetchingSites && ! this.props.isAlreadyOnSitesList ) ||
+			( userAlreadyConnected )
```

Comment:
Thanks! I get it now. Let me see if this is right:

* You're logged in to WordPress.com as a user who is linked to the site.
* You're logged in to the site, but with an account that _is not_ associated with your currently logged in WordPress.com account.
* You try to link. 💥  see a message

Yeah, this message is terrible:

> This site is already connected to a different WordPress.com user, you need to disconnect that user before you can connect another.

I don't think that really describes the same conditions we're talking about. Do we need a new notification?